samples/03.speech/h.select-audio-input-device/comprehensive.html

```diff
@@ -0,0 +1,246 @@
+    <!--
+      For simplicity and code clarity, we are using Babel and React from unpkg.com.
+    -->
+    <script crossorigin="anonymous" src="https://unpkg.com/babel-standalone@6/babel.min.js"></script>
+    <script crossorigin="anonymous" src="https://unpkg.com/react@16.8.6/umd/react.development.js"></script>
+    <script crossorigin="anonymous" src="https://unpkg.com/react-dom@16.8.6/umd/react-dom.development.js"></script>
+    <!-- Cognitive Services Speech Services adapter is only available in full bundle -->
+    <!--
+      This CDN points to the latest official release of Web Chat. If you need to test against Web Chat's latest bits, please refer to pointing to Web Chat's MyGet feed:
+      https://github.com/microsoft/BotFramework-WebChat#how-to-test-with-web-chats-latest-bits
+    -->
+    <!-- <script crossorigin="anonymous" src="https://cdn.botframework.com/botframework-webchat/latest/webchat.js"></script> -->
+    <script
+      crossorigin="anonymous"
+      src="https://github.com/microsoft/BotFramework-WebChat/releases/download/daily/webchat.js"
+    ></script>
+    <style>
+      html,
+      body {
+        height: 100%;
+      }
+
+      body {
+        margin: 0;
+      }
+
+      #app {
+        display: flex;
+        flex-direction: column;
+        height: 100%;
+        width: 100%;
+      }
+
+      #app .webchat {
+        overflow: hidden;
+      }
+
+      #app .app__audio-input-devices__header {
+        background-color: #090;
+        border-radius: 0 0 0 4px;
+        color: White;
+        font-size: 11px;
+        padding: 0px 4px 2px;
+        position: absolute;
+        right: 0;
+        top: 0;
+      }
+
+      #app .app__audio-input-devices {
+        background-color: #ffc;
+        border-radius: 4px;
+        border: solid 1px rgba(0, 0, 0, 0.1);
+        box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
+        flex-shrink: 0;
+        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans',
+          'Helvetica Neue', sans-serif;
+        font-size: 14px;
+        margin: 10px;
+        overflow: hidden;
+        padding: 10px;
+        position: relative;
+      }
+
+      #app .app__audio-input-devices__list {
+        list-style-type: none;
+        margin: 0;
+        padding: 0;
+      }
+
+      #app .app__audio-input-devices__list-item:not(:first-child) {
+        padding-top: 0.5em;
+      }
+
+      #app .app__audio-input-devices__device {
+        display: flex;
+      }
+
+      #app .app__audio-input-devices__radio-button {
+        margin-left: 0;
+        margin-right: 0.5em;
+      }
+    </style>
+  </head>
+  <body>
+    <div id="app" role="main"></div>
+    <script type="text/babel" data-presets="es2015,react,stage-3">
+      // Create a function to fetch the Cognitive Services Speech Services credentials.
+      // The async function created will hold expiration information about the token and will return cached token when possible.
```

Comment:
These explanations are lazy. The point of the `README.md` is to explain concepts there.